### PR TITLE
*: use API v2 of Alertmanager

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

**Comment by simonpasquier** [comment]
/hold 
it needs https://github.com/openshift/prometheus-operator/pull/51 first

**Comment by LiliC** [comment]
Do we want this merged for 4.4 or 4.5?

**Comment by simonpasquier** [comment]
/hold 

can wait after feature freeze

**Comment by LiliC** [comment]
@simonpasquier I am fine with merging it now as well, just meant if its for 4.5 then we need to aim it towards the master-4.5 branch.

**Comment by simonpasquier** [comment]
@LiliC ack, anyway keeping on hold since it conflicts with #624

**Comment by simonpasquier** [comment]
ping @LiliC @s-urbaniak this is now against `master-4.5`.

File: go.sum [hunk 1]
```diff
@@ -20,20 +20,15 @@ github.com/DataDog/datadog-go v2.2.0+incompatible/go.mod h1:LButxg5PwREeZtORoXG3
 github.com/Jeffail/gabs v1.1.1 h1:V0uzR08Hj22EX8+8QMhyI9sX2hwRu+/RJhJUmnwda/E=
```
Comment (LiliC) [inline go.sum]: Did you do a go mod tidy? :D

Comment (simonpasquier) [reply go.sum]: Probably, the `ci/prom/vendor` job should catch anything inconsistent.

**Comment by simonpasquier** [comment]
> Just curious, there were no changes between v1 and v2, why the jump to different versions?

v1 is deprecated and will be removed in a future version of Alertmanager. To have a seamless migration for users, we decided that for the alerts endpoint, v1 and v2 would use the same format.

**Comment by s-urbaniak** [comment]
@simonpasquier as master is now open for 4.5, we can fast-forward this there :-)

**Pull request merged**